.github/workflows/on-push-to-main.yaml

```diff
@@ -35,3 +39,17 @@ jobs:
           project_status: official
           project_stability: stable
           project_type: other
+
+      - uses: google-github-actions/release-please-action@v3
```

Comment:
I'm a little surprised to see this action being used in on-push-to-main, I thought it would replace something in `manual-release.yaml`.

So just to check my understanding, this will create a new "release" PR on our next push to main, and then whenever there is another push to main it will just keep updating that PR, and we decide when we want to do the release by actually merging that PR?

Is that their recommended workflow? Seems like it would also support just having a `manual-release` workflow that does nothing but call this action, and then the PR would just get created in a more on-demand fashion. Is that accurate?

If so I'm curious whether they talk about pros/cons between the two.

Reply:
Yes, that is absolutely the recommended/required workflow. The merge of the release PR *is* the manual release step. The action needs to be called both when our PRs are merged *and* when the release PR is merged, so the push-to-main trigger ensures it catches everything it needs to to both maintain the release PR and, when merging the release PR, to create the release in GH and trigger any potential `Publish` steps by setting the `release_created` flag. As far as I can tell, there isn't a supported workflow for triggering those stages manually.

Reply:
ack. let's demo this in our meeting today and then we can approve/merge.